File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    vector<pair<int,int>> candidate;
    for(int i=1;i<=int(sqrt(double(yellow)));i++) {
        if (yellow % i == 0) {
            candidate.push_back({i, yellow / i});
        }
    }
    for (auto [c1,c2]: candidate) {
        int need = 2 * c1 + 2 * c2 + 4;
        if (need == brown) {
            answer = {c1+2,c2+2};
        }
    }
    sort(answer.begin(), answer.end(), greater<>());
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math
def solution(brown, yellow):
    answer = []
    candidate = []
    for i in range(1, int(math.sqrt(yellow))+1):
        if yellow % i == 0:
            candidate.append((yellow // i,i))
    
    candidate.sort()
    for r,c in candidate:
        b = 2*r + 2*c + 4
        if b == brown:
            answer = [r+2,c+2]
            break
    return answer